sphinx_rtd_theme dependency can't be included in extras package

It appears that `setup.py` dependencies can't be in url format in released version of a package:

```sh
HTTPError: 400 Client Error: Invalid value for requires_dist. Error: Can't have direct dependency: "sphinx-rtd-theme @ git+https://github.com/augerai/sphinx_rtd_theme.git@bump-version#2ab38df0d303163e0e6c2bac80d907e9915000cb ; extra == 'all'" for url: https://upload.pypi.org/legacy/
Makefile:54: recipe for target 'release' failed
make: *** [release] Error 1

Exited with code exit status 2
```

We'll need to move these dependencies elsewhere.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -13,22 +13,26 @@ jobs:
       - run: sudo chown -R circleci:circleci /usr/local/lib/python3.7/site-packages
       - restore_cache:
           keys:
-            - a2ml-python-doc-deps-v2-{{ arch }}-3.7-{{ .Branch }}-{{ checksum "setup.py" }}
-            - a2ml-python-doc-deps-v2-{{ arch }}-3.7-{{ .Branch }}
-            - a2ml-python-doc-deps-v2-{{ arch }}-3.7
+            - a2ml-python-doc-deps-v3-{{ arch }}-3.7-{{ .Branch }}-{{ checksum "setup.py" }}-{{ checksum "docs/requirements.txt" }}
+            - a2ml-python-doc-deps-v3-{{ arch }}-3.7-{{ .Branch }}
+            - a2ml-python-doc-deps-v3-{{ arch }}-3.7
       - run:
           name: Install dependencies
           command: |
-            pip install ".[docs]"
+            virtualenv venv
+            source venv/bin/activate
+            make develop-docs
       - save_cache:
-          key: a2ml-python-doc-deps-v2-{{ arch }}-3.7-{{ .Branch }}-{{ checksum "setup.py" }}
+          key: a2ml-python-doc-deps-v3-{{ arch }}-3.7-{{ .Branch }}-{{ checksum "setup.py" }}-{{ checksum "docs/requirements.txt" }}
           paths:
-            # https://github.com/circleci/circleci-docs/pull/4214 ???
-            - "/usr/local/bin"
-            - "/usr/local/lib/python3.7/site-packages"
+            - "venv"
+            - "/home/circleci/.cache/pip"
       - run:
           name: Build docs
-          command: cd docs/ && make html
+          command: |
+            source venv/bin/activate 
+            cd docs/
+            make html
       - persist_to_workspace:
           root: docs/build
           paths: html

--- a/Makefile
+++ b/Makefile
@@ -44,6 +44,10 @@ clean:
 develop:
 	pip install -e ".[all]"
 
+develop-docs:
+	pip install -r docs/requirements.txt
+	pip install .
+
 init:
 	virtualenv .venv
 

--- a/README.md
+++ b/README.md
@@ -671,6 +671,12 @@ Running tests and getting test coverage:
 $ make test
 ```
 
+Running tests using docker:
+
+```sh
+$ make config docker-test
+```
+
 ## Authentication with A2ML
 Authentication with A2ML involves two parts. First, there is authentication between your client (whether it's the `a2ml` cli or the `a2ml` python API) and the service endpoint (either self-hosted or with Auger.AI). Second, there is authentication between the service endpoint and each provider. Note that in the case where you run A2ML locally, endpoint authentication is handled automatically. The table at the end of this section shows this in more detail.
 
@@ -678,7 +684,7 @@ Authentication with A2ML involves two parts. First, there is authentication betw
 You can login to the Auger.AI endpoint and provider with the `a2ml auth login` command.
 
 ```sh
-a2ml auth login
+$ a2ml auth login
 ```
 You will be prompted for your Auger service user and password. You can also download your Auger credentials as a auger.json file and put it to application folder.
 You can also put the path to auger.json in an environment variable called AUGER_CREDENTIALS_PATH OR a key inside AUGER.YAML.  
@@ -776,9 +782,15 @@ There are currently two top level directories with documentation.
 You can edit the files in here directly using <a href='https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html' target='_blank'>restructured text</a> syntax or you can add files for new sections.  Any new sections will need to have the path added in index.rst file.
 
 ### Building Documentation
-From inside the /docs directory run:
+First, install documentation-related dependencies:
 
+```sh
+make develop-docs
 ```
+
+Then, from inside the /docs directory run:
+
+```sh
 make html
 ```
 
@@ -786,7 +798,7 @@ This will create new files inside of the /build directory.  If you notice that y
 
 ### Viewing Documentation in the Browser
 
-```
+```sh
 open ./build/html/index.html
 ```
 
@@ -796,14 +808,14 @@ To release a new version the flow should be:
 
 2. Commit and push the changes above.
 
-```
+```sh
 git tag v<the-version> (for example: git tag v0.3.0)
 git push --tags
 ```
 
 3. verify circleci build passed and docker image tag exists:
 
-```
+```sh
 pip install -U a2ml==0.3.0
 docker pull augerai/a2ml:v0.3.0
 ```

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -1,0 +1,2 @@
+sphinx
+git+https://github.com/augerai/sphinx_rtd_theme.git@bump-version#2ab38df0d303163e0e6c2bac80d907e9915000cb'

--- a/setup.py
+++ b/setup.py
@@ -56,8 +56,7 @@ extras = {
         'wheel>=0.30.0,<0.31.0'
     ],
     'docs': [
-        'sphinx',
-        'sphinx_rtd_theme @ git+https://github.com/augerai/sphinx_rtd_theme.git@bump-version#2ab38df0d303163e0e6c2bac80d907e9915000cb'
+        'sphinx'
     ],
     'server': [
         'celery==4.4.0',

